extensions/in-memory/transfer-store-memory/src/main/java/org/eclipse/dataspaceconnector/transfer/store/memory/InMemoryTransferProcessStore.java

```diff
@@ -72,7 +73,9 @@ public String processIdForTransferId(String id) {
     @Override
     public void create(TransferProcess process) {
         writeLock(() -> {
-            process.transitionInitial();
+            if (process.getState() == TransferProcessStates.UNSAVED.code()) {
```

Comment:
I think this operation should be done in the `TransferProcessManager`, the store should just store.

Reply:
good catch